src/components/molecules/list.rs

```diff
@@ -17,436 +18,346 @@ use crate::{components::{atoms::{
+    let container_to_scroll = use_ref::<Option<Box<HtmlElement>>>(cx, || None);
+    let list_to_scroll = use_ref::<Option<Box<HtmlElement>>>(cx, || None);
+    let on_scroll = use_state(cx, || false);
+    let is_loading = use_state(cx, || cx.props.is_loading);
 
-            let close_style = r#"
-              cursor: pointer;
-              background: transparent;
-              border: 1px solid transparent;
-              display: flex;
-            "#;
+    let current_room = use_shared_state::<CurrentRoom>(cx).expect("Unable to load Current Room");
+    let replying_to = use_shared_state::<Option<ReplyingTo>>(cx).expect("Unable to load Replying to");
+    let timeline_thread = use_shared_state::<Option<TimelineThread>>(cx).expect("Unable to load timeline thread");
+    
+    let messages_list_thread = match timeline_thread.read().deref() {
+        Some(_)=> "messages-list--is-thread",
+        None => "messages-list--not-thread"
+    };
 
-            rsx!(
-                div {
-                    style: "
-                        padding: 12px;
-                        background: var(--background-modal);
-                        display: flex;
-                        justify-content: space-between;
-                        align-items: center;
-                        width: 100%;
-                        position: absolute;
-                        left: 0;
-                        border-radius: 16px 16px 0;
-                        z-index: 100;
-                    ",
-                    
-                    button {
-                        style: "{close_style}",
-                        onclick: move |_| {
-                            *timeline_thread.write() = None
-                        },
-                        Icon {
-                          stroke: "var(--icon-subdued)",
-                          icon: ArrowLeft
-                        }
-                      }
-                    p {
-                        style: "
-                            overflow: hidden;
-                            display: -webkit-box;
-                            -webkit-line-clamp: 1;
-                            -webkit-box-orient: vertical;
-                        ",
-                        "Hilo {title_thread}"
-                    }
-                    button {
-                        style: "{close_style}",
-                        onclick: move |_| {
-                            *timeline_thread.write() = None
-                        },
-                        Icon {
-                          stroke: "var(--icon-subdued)",
-                          icon: Close
-                        }
-                      }
-                }
-            )
+    use_effect(cx, (on_scroll,), |(_,)| {
+        let props = cx.props.to_owned();
+
+        if *on_scroll.get() && !cx.props.is_loading {
+            props.on_scroll.call(ListEvent {  });
         }
-        
+        async move {}
```

Comment:
Is this intentional?

Reply:
Yep, effects require something asynchronous to be defined, even if it is empty.